src/test/java/org/rulez/demokracia/pdengine/VoteTest.java

```diff
@@ -25,13 +25,28 @@ public void setUp() throws ReportedException {
 	@tested_operation("Cast vote")
 	@tested_behaviour("deletes the ballot with ballotId, so only one vote is possible with a ballot")
 	@Test
-	public void cast_vote_deletes_our_ballot() {
+	public void cast_vote_deletes_ballot_if_canVote_is_true() {
 		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
 		List<RankedChoice> theCastVote = new ArrayList<RankedChoice>();
 		Vote vote = getTheVote();
+		vote.canVote = true;
 		voteManager.castVote(adminInfo.voteId,ballot,theCastVote);
-		assertFalse(vote.ballots.contains(ballot));
-		
+		assertFalse(vote.ballots.contains(ballot));		
 	}
 
+	@tested_feature("Vote")
+	@tested_operation("Cast vote")
+	@tested_behaviour("works only if canVote is true")
+	@Test
+	public void cast_vote_does_not_delete_ballot_if_canVote_is_false() {
+		String ballot = voteManager.obtainBallot(adminInfo.voteId, adminInfo.adminKey);
+		List<RankedChoice> theCastVote = new ArrayList<RankedChoice>();
+		Vote vote = getTheVote();
+		vote.canVote = false;
+		
+		assertThrows(() -> voteManager.castVote(adminInfo.voteId,ballot,theCastVote)
+				).assertMessageIs("Cast vote does not get recorded");
```

Comment:
"This issue cannot be voted on on yet" would be a more appropriate message.

Reply:
Fixed